### PR TITLE
Improve 'expected a versioning field' error message

### PR DESCRIPTION
**Pull request by Pospelove**
Imagine you try to commit a port with missing or invalid version field in manifest. You'd get an error like this:
```
error: can't load port wasmedge
error: while loading /Users/leonidpospelov/projects/vcpkg/ports/wasmedge/vcpkg.json:
$ (a manifest): unexpected field 'version-sever', did you mean 'version-semver'?
$ (a manifest): expected a versioning field (example: version-string) 
```

However it seems that using of version-string is deprecated:
```
Use the version scheme "version" instead of "version-string" in port "wasmedge".
Use --skip-version-format-check to disable this check.
```

This PR makes error message consistent with lint warnings and shows all version-like fields available:

Before:
```
$ (a manifest): expected a versioning field (example: version-string) 
```

After:
```
$ (a manifest): expected a versioning field (one of 'version', 'version-semver', 'version-date') 
```

### CODE REVIEW

**Comment by ras0219-msft** [comment]
Thanks for the PR!

`version-string` is not deprecated -- however `version`, `version-date`, and `version-semver` provide a better experience for users and should be preferred IF the port's version matches those schemas.

**Comment by Pospelove** [comment]
@ras0219-msft I see. Thanks for your explanation! Should this PR be closed then or is it still valid?

**Comment by BillyONeal** [comment]
> @ras0219-msft I see. Thanks for your explanation! Should this PR be closed then or is it still valid?

I think this is valid but we need to add version-string to the list.

Also somehow we missed localizing this :(

File: src/vcpkg/versiondeserializers.cpp [hunk 1]
```diff
@@ -181,7 +181,16 @@ namespace vcpkg
...
+            r.add_generic_error(parent_type,
+                                "expected a versioning field (one of '",
+                                VERSION_RELAXED,
+                                "'",
+                                ", '",
+                                VERSION_SEMVER,
+                                "'",
+                                ", '",
+                                VERSION_DATE,
+                                "')");
```
Comment (BillyONeal) [inline src/vcpkg/versiondeserializers.cpp:193]:
```suggestion
            r.add_generic_error(parent_type,
              "expected a versioning field (one of version, version-date, version-semver, or version-string");
```

**Review by BillyONeal — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks!

**Pull request merged**